backend/controllers/mattresses.controller.ts

```diff
@@ -0,0 +1,22 @@
+import express, {Request, Response} from "express";
+import mongoose from "mongoose"
+import { Mattress } from "../models/mattress.model";
+import { getAllMattresses } from "../service/mattresses.service";
+import { createMattress } from "../service/mattresses.service";
+
+
+const MattressController = {
+    Index: async (req:Request, res: Response) => {
+        const mattresses = await getAllMattresses()
+        res.json({mattresses})
+    },
+
+    Create: async (req: Request, res: Response) => {
```

Comment:
I noticed you didn't use materials, when we create a mattress we would expect there to be an array of materials present. 

- having a size property means you now need to create the same mattress for 3ft, 4ft, 4ft6, 5ft. This is too repetitive and adds little value in this way. A good way to do this would be to have a size array of the available sizes.

- So im guessing type refers to something like 'Orthopedic'. Who controlling the available types - are they an enum or a user input. A good way to understand the issue with user input is what happens if a user sometimes spells the type as 'orthopedic' and other times spells it as 'orthopeadic' or makes a typo. We now lose the ability to group by type. 

- I am noticing a similar issue to the size property with cost and price. 

It might be worth going over how we want to design the Mattress object again.